rll/upgrade.sh

```diff
@@ -1,64 +1,129 @@
+  for retry_counter in {1..5}; do
+    # The auth information is updated on an upgrade.  Continue to source the
+    # auth file to grab the updated auth info once RightLink has restarted.
+    source ${RLL_SECRET}
+    new_installed_version=`curl --silent --show-error --request GET --header X-RLL-Secret:$RS_RLL_SECRET --globoff \
+      "http://127.0.0.1:$RS_RLL_PORT/rll/proc/version" || true`
+    if [[ $new_installed_version == $desired ]]; then
+      logger -t rightlink "New version active - $new_installed_version"
+      break
+    else
+      logger -t rightlink "Waiting for new version to become active."
+      sleep 2
+    fi
+  done
+  if [[ $new_installed_version != $desired ]]; then
+    logger -t rightlink "New version does not appear to be desired version: $new_installed_version"
+    exit 1
+  fi
+
+  # Report to audit entry that RightLink ugpraded.
+  instance_json=`curl --silent --show-error --request GET --header X-RLL-Secret:$RS_RLL_SECRET --globoff \
+    "http://127.0.0.1:$RS_RLL_PORT/api/sessions/instance"`
+  re='\{"rel":"self","href":"(/api/clouds/[0-9]+/instances/[0-9a-zA-Z]*)"\}'
+  if [[ $instance_json =~ $re ]]; then
+    instance_href="${BASH_REMATCH[1]}"
+    curl --silent --show-error --request POST --header X-RLL-Secret:$RS_RLL_SECRET --globoff \
+      "http://127.0.0.1:$RS_RLL_PORT/api/audit_entries" \
+      --data-urlencode "audit_entry[auditee_href]=${instance_href}" \
+      --data-urlencode "audit_entry[detail]=RightLink updated to '${new_installed_version}'" \
+      --data-urlencode 'audit_entry[summary]=RightLink updated'
+  else
+    logger -t rightlink "unable to obtain instance href for audit entries"
+  fi
+  exit 0
+}
+
+source ${RLL_SECRET}
+
+# Detemine bin_path
+rl_bin=`curl --silent --show-error --request GET --header X-RLL-Secret:$RS_RLL_SECRET --globoff \
+  "http://127.0.0.1:$RS_RLL_PORT/rll/proc/bin_path"`
+prefix_url='https://rightlinklite.rightscale.com/rll'
+
+# Determine current version of rightlink
+current_version=`curl --silent --show-error --request GET --header X-RLL-Secret:$RS_RLL_SECRET --globoff \
+  "http://127.0.0.1:$RS_RLL_PORT/rll/proc/version"`
+
+if [ -z $current_version ]; then
   echo "Can't determine current version of RLL"
-  $RLBIN -version
   exit 1
 fi
 
-# Fetch information about what we should become, the "upgrade" file consists
-# of lines with "current_version:new_version"
-match=`curl -sS --retry 3 $prefix/self_upgrade2/upgrades | egrep "^${current}:" || true`
-re="^${current}: *([^ ]*)"
+# Fetch information about what we should become. The "upgrades" file is obtained
+# using the name of the current version.  The file consists of lines formatted as
+# "current_version: upgradeable_new_version"
+# If the "upgrades" file does not exist, no upgrade is done.
+re="^\s*${current_version}\s*:\s*(\S+)\s*$"
+match=`curl --silent --show-error --retry 3 ${prefix_url}/${current_version}/upgrades | egrep ${re} || true`
```

Comment:
I don't see how this is going to work. We need one global upgrades file, not one per branch. I guess technically it could work, but it would be a nightmare to maintain. I believe the upgrade file should not be something that is automatically pushed via travis but a file at the top-level of the rll tree.